src/news/dev-updates/may-16-23.md

```diff
@@ -0,0 +1,56 @@
+
+## New Features and Enhancements🆕
+
+- **Environment-Only Configuration:** You can now run an Iroha node using only environment variables, without a configuration file. The team introduced additional environment variable settings covering all required configuration fields, so the `--config flag` is no longer mandatory. (Iroha will still default to client.toml if present, but it’s no longer required.) This change simplifies deployments and scripting, making Iroha more flexible to run in containerized or cloud environments. As part of this update, an integration test suite for the Iroha CLI was added to ensure these new config options work reliably.
+
+- **Test Network Improvements:** Significant enhancements were made to the in-house test network framework to improve modularity and control. Each test peer in the network now gets a human-readable mnemonic name (e.g. spooky_tiger) used for its directory and logs, making debugging easier. The test network configuration is now layered, allowing easier customization and overrides of defaults. Developers can also generate network topologies deterministically by providing seeds, ensuring reproducible test scenarios. New utility functions have been introduced: `NetworkPeer::start_checked` will start a node and immediately check if it fails (catching startup failures early), and a standalone `once_block_sync` function can pause tests until all peers reach a certain block height (with failure detection). Additionally, a unified Program utility now resolves the correct binary (irohad or the maynewer iroha binary) in test environments. Collectively, these changes make writing and running integration tests more robust and developer-friendly.
+
+- **WASM Fuel Configuration for Tests:** A new helper `NetworkBuilder::with_wasm_fuel()` was added to the testing toolkit. This lets developers configure the WASM execution fuel (instruction limits for smart contracts) in test networks, which helps in simulating and validating smart contract behavior under different resource limits. It’s a small addition but useful for fine-tuning how smart contracts run in a controlled environment.
+
+## Bug Fixes and Issue Resolutions🐞
+
+- **WASM Event Coverage:** The team closed an issue related to WebAssembly smart contracts not properly declaring all possible events they can emit. The WASM executor now declares the full union of possible events, ensuring that event handling logic is aware of every event type a contract might produce. This fix improves correctness and transparency in Iroha’s contract system.
+
+- **Deduplicating WASM Blobs:** Another fix addressed a problem of duplicate WASM binaries in the system. The resolution ensures that WASM blobs are not redundantly stored or processed (“`Dedup wasm blobs`”), which reduces unnecessary memory and storage overhead. This optimization makes Iroha’s handling of smart contract modules more efficient.
```

Comment:
This remains unresolved. Please distinguish between "closed as duplicate" and "closed as completed".